### PR TITLE
Make `lifecycleOrNull` public.

### LINKED ISSUE #734: Kotlin: Make Context.lifecycleOrNull public

### PR DESCRIPTION
**Pull request by zach-klippenstein**
Closes #734.

### CODE REVIEW

File: kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/Lifecycles.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+
+/**
+ * The [Lifecycle] for this context, or null if one can't be found.
+ *
+ * @implNote We keep all this very forgiving because we're just using it to keep some logging
```
Comment (rjrjr) [inline kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/Lifecycles.kt]: Should drop the `implNote` entirely.

Comment (zach-klippenstein) [reply kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/Lifecycles.kt]: It won't show up in rendered kdoc, if that's what you're concerned about.

Comment (rjrjr) [reply kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/Lifecycles.kt]: Neat to know, but it's misleading. IIRC we use it for more than just that original reason these days.

Comment (zach-klippenstein) [reply kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/Lifecycles.kt]: Removed.

**Comment by zach-klippenstein** [comment]
Swift build errored, but this doesn't contain any swift changes and all the Android stuff is green, so I'm going to go ahead and merge.

**Pull request merged**